### PR TITLE
Add function to export a daterange-bounded list of posts, ordered by karma

### PR DESCRIPTION
**Pull request by colophonemes**

### CODE REVIEW

File: packages/lesswrong/server/scripts/exportPostDetails.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+Vulcan.exportPostDetails = wrapVulcanAsyncScript(
+  'exportPostDetails',
+  async ({selector, outputDir, outputFile = 'post_details.csv'}) => {
+    if (!outputDir) throw new Error(`you must specify an output directory (hint: {outputDir})`)
+    const documents = getPosts(selector)
+    let c = 0
+    const count = documents.count()
+    const rows = []
+    for (let post of documents) {
+      // SD: this makes things horribly slow, but no idea how to do a more efficient join query in Mongo
```
Comment (Discordius) [inline packages/lesswrong/server/scripts/exportPostDetails.js]: The answer is aggregate queries, but I guess it seems fine without it.

File: packages/lesswrong/server/scripts/exportPostDetails.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        id: post._id,
+        user_id: post.userId,
+        title: post.title,
+        slug: post.slug,
+        karma: post.baseScore,
+        community: !!post.meta,
+        frontpage_date: post.frontpageDate,
+        posted_at: post.postedAt,
+        created_at: post.createdAt,
+        url: `https://forum.effectivealtruism.org/posts/${post._id}/${post.slug}`
```
Comment (Discordius) [inline packages/lesswrong/server/scripts/exportPostDetails.js]: Hmm, this seems like we should somehow abstract it. I think we already have a Root-URL variable in the settings file, so this should be pretty straightforward.

Actually, you probably just want to use the `Posts.getPageUrl(...)` function, which has an `absolute` flag you can pass to get the absolute URL.

**Comment by Discordius** [comment]
Obviously, would be cool if you address the requested changes, but also seems fine for us to copy this branch and address them ourselves (though this removes the benefit of avoiding future merge-conflicts a bit).

**Comment by colophonemes** [comment]
@Discordius / @jimrandomh – I'm not sure I'm inclined to make this better with aggregate queries but I'd love to see it merged. If the offer of you guys merging it and us dealing with the conflicts later is still on the table let me know, otherwise I'll close this and just run it locally.

**Comment by jpaddison3** [comment]
@Discordius ready for rereview

**Comment by Discordius** [comment]
I do still think it's important to use the Rooturl variable instead of hardcoding the EA-Forum

**Comment by jpaddison3** [comment]
_Sigh_ yes that does seem important. I think Sam may have had a cached thought about it being ready other than the aggregate queries.

**Comment by jpaddison3** [comment]
@Discordius ready for rereview

**Pull request merged**